UPGRADE-7.0.md

```diff
@@ -301,6 +376,58 @@ Serializer
  * Add argument `$context` to `NormalizerInterface::supportsNormalization()` and `DenormalizerInterface::supportsDenormalization()`
  * Remove Doctrine annotations support in favor of native attributes
  * Remove the annotation reader parameter from the constructor of `AnnotationLoader`
+ * The following Normalizer classes have become final, use decoration instead of inheritance:
+   * `ConstraintViolationListNormalizer`
+   * `CustomNormalizer`
+   * `DataUriNormalizer`
+   * `DateIntervalNormalizer`
+   * `DateTimeNormalizer`
+   * `DateTimeZoneNormalizer`
+   * `GetSetMethodNormalizer`
+   * `JsonSerializableNormalizer`
+   * `ObjectNormalizer`
+   * `PropertyNormalizer`
+
+   *Before*
+   ```php
+   use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+
+   class TopicNormalizer extends ObjectNormalizer
+   {
+       // ...
+
+       public function normalize($topic, string $format = null, array $context = []): array
+       {
+           $data = parent::normalize($topic, $format, $context);
+
+           // ...
+       }
+   }
+   ```
+
+   *After*
+   ```php
+   use Symfony\Component\DependencyInjection\Attribute\Autowire;
+   use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+   use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+   class TopicNormalizer implements NormalizerInterface
+   {
+       public function __construct(
+           #[Autowire(service: 'serializer.normalizer.object')] private NormalizerInterface&DenormalizerInterface $objectNormalizer,
```

Comment:
we could also alias `NormalizerInterface&DenormalizerInterface` to `serializer.normalizer.object` BTW

Reply:
I'm not sure about that. Most normalizers fullfill this type, and which one you want depens on the use case (although object normalizer is probably the 80% use-case).